canonical/vanilla-framework
# Table layout utilities classes also apply to descendant elements; add table layout example docs

## Done

- Updated the `u-table-layout--auto` and `u-table-layout--fixed` utility classes to apply to nested / descended `table` elements as well as elements with those classes directly applied.
- Added example docs for the table layout utility settings. This includes:
  - Default (fixed) table layout (`$table-layout-fixed: true` and no utility class applied directly to a table)
  - Automatic table layout (`$table-layout-fixed: true` and `u-table-layout--auto` applied directly to a table)
  - Automatic table layout, nested (`$table-layout-fixed: true` and `u-table-layout--auto` applied to an ancestor element of a table 

Fixes #2374 , [WD-10925](https://warthogs.atlassian.net/browse/WD-10925) 

## QA

- Open [demo](https://vanilla-framework-5070.demos.haus/docs/settings/table-layout)
- Verify that the example tables in the table layout docs page appear as they do in the attached screenshots.
    - Tables in the table layout utility documentation page should have `table-layout: auto` applied.
- Review updated documentation:
  - Docs -> Settings -> Table layout

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)

## Screenshots
![image](https://github.com/canonical/vanilla-framework/assets/46915153/5a8545bd-2845-458d-9005-9e1f98d375fc)

[WD-10925]: https://warthogs.atlassian.net/browse/WD-10925?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
